internal/grpc/services/userprovider/userprovider.go

```diff
@@ -59,7 +70,37 @@ func parseConfig(m map[string]interface{}) (*config, error) {
 	return c, nil
 }
 
-func getDriver(c *config) (user.Manager, error) {
+// getDriverPlugin fetches the runtime driver from the plugins package
+func getDriverPlugin(c *config) (user.UserManager, error) {
+	sym, err := c.load()
+	if err != nil {
+		return nil, err
+	}
+
+	// assert the loaded plugin into required interface
+	manager, ok := sym.(user.UserManager)
+	if !ok {
+		return nil, fmt.Errorf("could not assert the loaded plugin")
+	}
+
+	pluginConfig := filepath.Base(c.Driver)
```

Comment:
I'm using `filepath.Base` to retrieve the actual driver name from the config. Since, whenever we want load a binary, we're providing the absolute path (something like `/home/path/to/binary`), due to which we would not be able to access the plugin configuration via `c.Drivers[c.Driver]` (this would just result in an empty map) because, `c.Driver` would be `/home/path/to/binary` as opposed to just the driver name. So in order to access the plugin configuration I had to use `filepath.Base`

Reply:
That is okay! We can think of having some aliasing in the future.